opensearch-service-domain-cdk/test/stack-composer.test.ts

```diff
@@ -0,0 +1,225 @@
+import {App} from "aws-cdk-lib";
+import {StackComposer} from "../lib/stack-composer";
+import {Template} from "aws-cdk-lib/assertions";
+
+test('Test missing domain name throws error', () => {
+
+    const app = new App({
+        context: {
+            domainName: ""
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+       env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test missing engine version throws error', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: ""
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test invalid engine version format throws error', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: "OpenSearch_1.3"
+        }
+    })
+
+    const createStackFunc = () => new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    expect(createStackFunc).toThrowError()
+})
+
+test('Test ES engine version format is parsed', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: "ES_7.10"
+        }
+    })
+
+    const openSearchStacks =  new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    const domainStack = openSearchStacks.stacks.filter((s) => s.stackName === "opensearchDomainStack")[0]
+    const domainTemplate = Template.fromStack(domainStack)
+    domainTemplate.resourceCountIs("AWS::OpenSearchService::Domain", 1)
+})
+
+test('Test OS engine version format is parsed', () => {
+
+    const app = new App({
+        context: {
+            engineVersion: "OS_1.3"
+        }
+    })
+
+    const openSearchStacks =  new StackComposer(app, {
+        env: {account: "test-account", region: "us-east-1"}
+    })
+
+    const domainStack = openSearchStacks.stacks.filter((s) => s.stackName === "opensearchDomainStack")[0]
+    const domainTemplate = Template.fromStack(domainStack)
+    domainTemplate.resourceCountIs("AWS::OpenSearchService::Domain", 1)
+})
+
+test('Test access policy is parsed for proper array format', () => {
+
+    const app = new App({
+        context: {
+            accessPolicies: {"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"AWS":"arn:aws:iam::123456789123:user/test-user"},"Action":"es:ESHttp*","Resource":"arn:aws:es:us-east-1:123456789123:domain/test-os-domain/*"},
+                    {"Effect":"Allow","Principal":{"AWS":"arn:aws:iam::123456789123:user/test-user2"},"Action":"es:ESHttp*","Resource":"arn:aws:es:us-east-1:123456789123:domain/test-os-domain/*"}]}
```

Comment:
This would be more readable on more lines - even if you just put the start/end array []s on their own lines, to show that each line was an array element